src/components/createBook/CreateBook.tsx

```diff
@@ -1,26 +1,167 @@
+
+import { ChangeEvent, useEffect, useState } from 'react'
+
+import { MdAddCircleOutline as AddIcon } from 'react-icons/md'
+import { useNavigate } from 'react-router-dom'
+import Select, { MultiValue } from 'react-select'
+
 import noCover from '../../assets/no-cover.png'
+import AuthorResponse from '../../interfaces/AuthorResponse'
+import Book from '../../interfaces/Book'
+import { getAuthors } from '../../services/author.service'
+import { createBook } from '../../services/book.service'
+import CreateAuthorModal from '../createAuthorModal/CreateAuthor'
 import './CreateBook.css'
 
+const initialBook: Book = { Title: '', ISBN: '', Quantity: 0, AuthorsIds: [], Description: '', PublishDate: '' }
+
 function CreateBook() {
+  const [ authorList, setAuthorList ] = useState<AuthorResponse[]>([])
+  const [ selectedAuthors, setSelectedAuthors ] = useState<AuthorResponse[]>([])
+  const [ isAuthorsChanged, setIsAuthorsChanged ] = useState(false)
+  const [ cover, setCover ] = useState(noCover)
+  const [ show, setShow ] = useState(false)
+  const [ titleErrorMessage, setTitleErrorMessage ] = useState ('')
+  const [ ISBNErrorMessage, setISBNErrorMessage ] = useState ('')
+  const [ quantityErrorMessage, setQuantityErrorMessage ] = useState ('')
+  const [ requestCover, setRequestCover ] = useState<Blob>(new Blob)
+  const [ book, setBook ] = useState<Book>(initialBook)
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    fetchAuthors()
+  }, [ isAuthorsChanged ])
+
+  const fetchAuthors = () => {
+    getAuthors().then(response => {
+      setAuthorList(response.data)
+    }).catch(error => {console.error(error)})
+  }
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const files = event.target?.files
+    const reader = new FileReader()
+    if(files!= null) {
+      reader.readAsDataURL(files[0])
+      setRequestCover(files[0])
+      reader.onloadend = function () {
+        const base64data = reader.result
+        if(base64data)
+          setCover(base64data as string)
+      }
+    }
+  }
+
+  const handleSelectedAuthorsChange = (authorsData: MultiValue<AuthorResponse>) => {
+    setSelectedAuthors(authorsData as AuthorResponse[])
+  }
+
+  const create = () => {
```

Comment:
Use more descriptive names for create and validate, I'm not sure what exactly you are validating here